crates/nu-cli/src/commands/split/row.rs

```diff
@@ -35,41 +35,45 @@ impl WholeStreamCommand for SubCommand {
         args: CommandArgs,
         registry: &CommandRegistry,
     ) -> Result<OutputStream, ShellError> {
-        split_row(args, registry)
+        split_row(args, registry).await
     }
 }
 
-fn split_row(args: CommandArgs, registry: &CommandRegistry) -> Result<OutputStream, ShellError> {
+async fn split_row(
+    args: CommandArgs,
+    registry: &CommandRegistry,
+) -> Result<OutputStream, ShellError> {
     let registry = registry.clone();
-    let stream = async_stream! {
-        let name = args.call_info.name_tag.clone();
-        let (SplitRowArgs { separator }, mut input) = args.process(&registry).await?;
-        while let Some(v) = input.next().await {
-            if let Ok(s) = v.as_string() {
-                let splitter = separator.item.replace("\\n", "\n");
-                trace!("splitting with {:?}", splitter);
-                let split_result: Vec<_> = s.split(&splitter).filter(|s| s.trim() != "").collect();
+    let name = args.call_info.name_tag.clone();
+    let (SplitRowArgs { separator }, mut input) = args.process(&registry).await?;
 
-                trace!("split result = {:?}", split_result);
+    let mut values_vec_deque = VecDeque::new();
 
-                for s in split_result {
-                    yield ReturnSuccess::value(
-                        UntaggedValue::Primitive(Primitive::String(s.into())).into_value(&v.tag),
-                    );
-                }
-            } else {
-                yield Err(ShellError::labeled_error_with_secondary(
-                    "Expected a string from pipeline",
-                    "requires string input",
-                    name.span,
-                    "value originates from here",
-                    v.tag.span,
-                ));
+    while let Some(v) = input.next().await {
+        if let Ok(s) = v.as_string() {
+            let splitter = separator.item.replace("\\n", "\n");
+            trace!("splitting with {:?}", splitter);
+            let split_result: Vec<_> = s.split(&splitter).filter(|s| s.trim() != "").collect();
+
+            trace!("split result = {:?}", split_result);
+
+            for s in split_result {
+                values_vec_deque.push_back(
+                    UntaggedValue::Primitive(Primitive::String(s.into())).into_value(&v.tag),
+                );
             }
+        } else {
+            return Err(ShellError::labeled_error_with_secondary(
+                "Expected a string from pipeline",
+                "requires string input",
+                name.span,
+                "value originates from here",
+                v.tag.span,
+            ));
         }
-    };
+    }
 
-    Ok(stream.to_output_stream())
+    Ok(futures::stream::iter(values_vec_deque).to_output_stream())
```

Comment:
There's a way to stream this also, though a little tricky.

`Ok(input.map(|item| **create the vecdeque for the current item**).flatten().to_output_stream())`